webextension/js/popup.js

```diff
@@ -1,150 +1,335 @@
+      }).then(() => {
+        window.close();
+      });
+    });
+  },
+
+  // This method is called when the panel is shown.
+  prepare() {
+    let identity = Logic.currentIdentity();
+
+    // Populating the panel: name and icon
+    document.getElementById("container-info-name").innerText = identity.name;
+
+    let icon = document.getElementById("container-info-icon");
+    icon.setAttribute("data-identity-icon", identity.image);
+    icon.setAttribute("data-identity-color", identity.color);
+
+    // Show or not the has-tabs section.
+    for (let trHasTabs of document.getElementsByClassName("container-info-has-tabs")) {
+      trHasTabs.hidden = !identity.hasHiddenTabs && !identity.hasOpenTabs;
+    }
+
+    const hideShowIcon = document.getElementById("container-info-hideorshow-icon");
+    hideShowIcon.src = identity.hasHiddenTabs ? CONTAINER_UNHIDE_SRC : CONTAINER_HIDE_SRC;
 
-browser.runtime.sendMessage({method: "queryIdentities"}).then(identities => {
-  console.log('queryIdentities');
-  let fragment = document.createDocumentFragment();
-
-  identities.forEach(identity => {
-    console.log('identities.forEach');
-    let tr = document.createElement("tr");
-    fragment.appendChild(tr);
-    tr.classList.add("container-panel-row");
-    tr.classList.add("clickable");
-    tr.setAttribute("data-identity-cookie-store-id", identity.userContextId);
-    tr.innerHTML = `
-      <td>
-        <div class="userContext-icon open-newtab"
-          data-identity-icon="${identity.image}"
-          data-identity-color="${identity.color}">
-        </div>
-      </td>
-      <td class="open-newtab">${identity.name}</td>
-      <td class="info">&gt;</td>`;
-
-    tr.addEventListener("click", e => {
-      if (e.target.matches(".open-newtab")) {
-        browser.runtime.sendMessage({
-          method: "showTabs",
-          userContextId: identity.userContextId
-        }).then(() => {
-          return browser.runtime.sendMessage({
-            method: "openTab",
-            userContextId: identity.userContextId,
+    const hideShowLabel = document.getElementById("container-info-hideorshow-label");
+    hideShowLabel.innerText = identity.hasHiddenTabs ? "Show these container tabs" : "Hide these container tabs";
+
+    // Let's remove all the previous tabs.
+    for (let trTab of document.getElementsByClassName("container-info-tab")) {
+      trTab.remove();
+    }
+
+    // Let's retrieve the list of tabs.
+    return browser.runtime.sendMessage({
+      method: "getTabs",
+      userContextId: identity.userContextId,
+    }).then(tabs => {
+      log('browser.runtime.sendMessage getTabs, tabs: ', tabs);
+      // For each one, let's create a new line.
+      let fragment = document.createDocumentFragment();
+      for (let tab of tabs) {
+        let tr = document.createElement("tr");
+        fragment.appendChild(tr);
+        tr.classList.add("container-info-tab");
+        tr.classList.add("clickable");
```

Comment:
Re: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

It looks like we can pass multiple classes to the `classList` by passing multiple values:

```js
tr.classList.add("container-info-tab", "clickable");
```